File: study/ts/script/data/ActivitySevenDaySprintInfoData.ts
import { BaseData } from './BaseData';
import { G_UserData, G_ServerTime } from '../init';
import { Lang } from '../lang/Lang';
import { TimeLimitActivityConst } from '../const/TimeLimitActivityConst';
import CommonConst from '../const/CommonConst';
import { FunctionCheck } from '../utils/logic/FunctionCheck';
import { TimeConst } from '../const/TimeConst';
let schema = {};
schema['id'] = [
    'number',
    0
];
schema['sprint_type'] = [
    'number',
    0
];
schema['name'] = [
    'string',
    ''
];
schema['function_id'] = [
    'number',
    0
];
schema['is_work'] = [
    'number',
    0
];
schema['open_day'] = [
    'number',
    0
];
schema['over_day'] = [
    'number',
    0
];
schema['over_view_day'] = [
    'number',
    0
];
schema['order'] = [
    'number',
    0
];
schema['activity_content_text'] = [
    'string',
    ''
];
export interface ActivitySevenDaySprintInfoData {
getId(): number
setId(value: number): void
getLastId(): number
getSprint_type(): number
setSprint_type(value: number): void
getLastSprint_type(): number
getName(): string
setName(value: string): void
getLastName(): string
getFunction_id(): number
setFunction_id(value: number): void
getLastFunction_id(): number
getIs_work(): number
setIs_work(value: number): void
getLastIs_work(): number
getOpen_day(): number
setOpen_day(value: number): void
getLastOpen_day(): number
getOver_day(): number
setOver_day(value: number): void
getLastOver_day(): number
getOver_view_day(): number
setOver_view_day(value: number): void
getLastOver_view_day(): number
getOrder(): number
setOrder(value: number): void
getLastOrder(): number
getActivity_content_text(): string
setActivity_content_text(value: string): void
getLastActivity_content_text(): string
}
export class ActivitySevenDaySprintInfoData extends BaseData {

    public static schema = schema;
    constructor (properties?) {
        super(properties);
        if (properties) {
            this.updateData(properties);
        }
    }
    public clear () {
    }
    public reset () {
    }
    public updateData (data) {
        this.setProperties(data);
    }
    public getType () {
        return this.getSprint_type();
    }
    public getDescription () {
        let des = this.getActivity_content_text();
        let actType = this.getSprint_type();
        let finalDes = des;
        if (actType == TimeLimitActivityConst.ACTIVITY_TYPE_GUILD_SPRINT) {
            finalDes = Lang.getTxt(des, { num: this.getOver_day() }) || '';
        }
        return finalDes;
    }
    public isActivityCompetitionTimeEnd () {
        let openDays = G_UserData.getBase().getOpenServerDayNum();
        return openDays >= this.getOver_day();
    }
    public getActivityStartEndTime () {
        let openServerTime = G_UserData.getBase().getServer_open_time();
        let openZeroTime = G_ServerTime.secondsFromZero(openServerTime, TimeConst.RESET_TIME_SECOND);
        let startTime = openZeroTime + (this.getOpen_day() - 1) * (3600 * 24);
        let endTime = openZeroTime + (this.getOver_view_day() - 1) * (3600 * 24);
        let competitionEndTime = openZeroTime + (this.getOver_day() - 1) * (3600 * 24);
        return [
            startTime,
            endTime,
            competitionEndTime
        ];
    }
    public isActivityOpen () {
        if (!this.isOpenCheckFunc()) {
            return false;
        }
        let openDays = G_UserData.getBase().getOpenServerDayNum();
        if (openDays >= this.getOpen_day() && openDays < this.getOver_view_day()) {
            return true;
        }
        return false;
    }

    private isOpenCheckFunc(){
        return this.getIs_work() == CommonConst.TRUE_VALUE && (this.getFunction_id() == 0 && true || FunctionCheck.funcIsOpened(this.getFunction_id())[0]);
    }
}
 ActivitySevenDaySprintInfoData;

File: study/lua/src/app/data/ActivitySevenDaySprintInfoData.lua

local BaseData = require("app.data.BaseData")
local ActivitySevenDaySprintInfoData = class("ActivitySevenDaySprintInfoData", BaseData)

local schema = {}
schema["id"] = {"number", 0}     -- id
schema["sprint_type"] = {"number", 0} -- 类型
schema["name"] = {"string", ""} -- 活动名称
schema["function_id"] = {"number", 0} -- 开启等级
schema["is_work"] = {"number", 0} -- 是否生效
schema["open_day"] = {"number", 0}  -- 开启时间
schema["over_day"] = {"number", 0} -- 结束时间
schema["over_view_day"] = {"number", 0} -- 结束后显示时间
schema["order"] = {"number", 0} -- 页签排序
schema["activity_content_text"] = {"string", ""}-- 活动内容文本


ActivitySevenDaySprintInfoData.schema = schema

function ActivitySevenDaySprintInfoData:ctor(properties)
	ActivitySevenDaySprintInfoData.super.ctor(self)

	if properties then	
		self:updateData(properties)
	end
end

function ActivitySevenDaySprintInfoData:clear()
	
end

function ActivitySevenDaySprintInfoData:reset()
	
end

function ActivitySevenDaySprintInfoData:updateData(data)
	self:setProperties(data)
end


function ActivitySevenDaySprintInfoData:getType()
	return self:getSprint_type()
end

function ActivitySevenDaySprintInfoData:getDescription()
	local TimeLimitActivityConst = require("app.const.TimeLimitActivityConst")
	local des = self:getActivity_content_text()
	local actType = self:getSprint_type()
	local finalDes = des
	if actType == TimeLimitActivityConst.ACTIVITY_TYPE_GUILD_SPRINT  then
		finalDes = Lang.getTxt(des,{num = self:getOver_day()}) or ""
	end  
	return finalDes
end

function ActivitySevenDaySprintInfoData:isActivityCompetitionTimeEnd()
	local CommonConst = require("app.const.CommonConst")
	local openDays = G_UserData:getBase():getOpenServerDayNum()
	return openDays >= self:getOver_day() 
end
 
function ActivitySevenDaySprintInfoData:getActivityStartEndTime()
	local CommonConst = require("app.const.CommonConst")
	local TimeConst = require("app.const.TimeConst")
	local openServerTime = G_UserData:getBase():getServer_open_time()
	local openZeroTime = G_ServerTime:secondsFromZero(openServerTime,TimeConst.RESET_TIME_SECOND)
	local startTime = openZeroTime + (self:getOpen_day()-1) * (3600 *24)
	local endTime = openZeroTime + (self:getOver_view_day()-1) * (3600 *24)
	local competitionEndTime = openZeroTime + (self:getOver_day()-1) * (3600 *24)
	return startTime,endTime,competitionEndTime
end

function ActivitySevenDaySprintInfoData:isActivityOpen()
	local FunctionCheck = require("app.utils.logic.FunctionCheck")
	local CommonConst = require("app.const.CommonConst")

	local isOpenCheckFunc = function()
		return self:getIs_work() == CommonConst.TRUE_VALUE and
			( self:getFunction_id()  == 0 and true or FunctionCheck.funcIsOpened(self:getFunction_id() ) )
	end
	if not isOpenCheckFunc() then
		return false
	end

	local openDays = G_UserData:getBase():getOpenServerDayNum()

	if openDays >= self:getOpen_day()  and openDays < self:getOver_view_day() then
		return true
	end
	return false
end

return ActivitySevenDaySprintInfoData